### ISSUE TITLE
[api-extractor] Support returning results instead of writing to file

### ISSUE THREAD

**Issue by chriskrycho** [microsoft/rushstack#2948]
## Summary

It would be nice, when building a custom pipeline on top of API Extractor, to be able to simply work with the resulting data directly instead of having it be written to disk and then read in again in a separate step. If nothing else, for sufficiently large projects, the I/O cost alone of writing and reading can eat up a lot of time; but more importantly it's much easier to work with the data by receiving it directly than having to make sure multiple tools are configured just so to get the data piped around correctly.

(It's possible I've missed a way to do this, but I've read the docs and API docs and don't see anything that suggests it's possible, and indeed quite the contrary!)

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/api-extractor` version? | 7.x |
| Operating system? | Mac/Linux |
| API Extractor scenario? | all of: reporting (.api.md) / rollups (.d.ts) / docs (.api.json) |
| Would you consider contributing a PR? | Yes |
| TypeScript compiler version? | 4.4 |
| Node.js version (`node -v`)? | 14.17.0 |

**Comment by octogonz** [comment]
Is this a dupe of https://github.com/microsoft/rushstack/issues/1010 ?

**Comment by chriskrycho** [comment]
I don’t think so; they’re related but not identical. (Both would be good to have!) Here I’m requesting that the in-between `.api.json` artifacts be available programmatically, whereas that one is asking for the type definitions which end up in `.d.ts` files from a rollup.
